cmake/extensions/CMakeLists.txt

```diff
@@ -122,10 +122,13 @@ add_python_extension(_testmultiphase REQUIRES IS_PY3 SOURCES _testmultiphase.c)
 # debug tool to trace memory blocks allocated by Python
 add_python_extension(_tracemalloc BUILTIN REQUIRES IS_PY3 SOURCES hashtable.c _tracemalloc.c)
 add_python_extension(_weakref BUILTIN REQUIRES IS_PY3 SOURCES _weakref.c)
-math(EXPR _limited_api_version "${PY_VERSION_MAJOR} * 100 + ${PY_VERSION_MINOR}")
+if(NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
```

Comment:
Should probably be something like : 
```
set(_limited_definition)
if(NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
    math(EXPR _limited_api_version "${PY_VERSION_MAJOR} * 100 + ${PY_VERSION_MINOR}")
    set(_limited_definition "Py_LIMITED_API=0x${_limited_api_version}0000")
endif()
```

That said, the approach should still be tweaked to handle the case of Visual Studio where `CMAKE_CONFIGURATION_TYPES` is defined

Reply:
@dbrnz I had an other look at this, and if `Py_LIMITED_API` must not be defined when doing `Debug` build, relying on a newer version of CMake supporting generator expression would be the way to go. 

That way, we will be able to do something like this:

```
target_compile_definition(xxlimited PRIVATE $<$<NOT:$<CONFIG:Debug>>:${_limited_definition}>)
```